src/internal/connector/onedrive/item.go

```diff
@@ -108,18 +108,18 @@ func downloadItemMeta(
 // TODO: @vkamra verify if var session is the desired input
 func driveItemWriter(
 	ctx context.Context,
-	rh RestoreHandler,
+	nicu NewItemContentUploader,
```

Comment:
Could you please explain again how this is better than `rh RestoreHandler`? Applies to all similar changes in this PR. I feel like it's a lot of cognitive load tracking all the new interfaces.  

If the added benefit is mocking, i think we could have that effect with `RestoreHandler` itself, right?

Reply:
Lots of ink is already spilled on the topic, so I won't rehash much.  Here's a quick read for it: https://www.integralist.co.uk/posts/go-interfaces/#keep-interfaces-small.  And here: https://betterprogramming.pub/a-comprehensive-guide-to-interfaces-in-go-6a945b795db#1fbe.  

Long and short of the philosophy: a function should request only the interface it uses, not the interface that the caller is able to supply.  If it's bad to import packages that aren't utilized, and it's bad to declare variables that aren't used, shouldn't it be bad to declare methods that aren't called?

Reply:
Thanks for the explanation! Those were excellent articles. I agree with your points, just feel that it can limit readability a bit. Maybe that's just me being a noob go enthusiast. I'll get there.

Reply:
>  Maybe that's just me being a noob go enthusiast.

Nah, this is one of those bikeshedding, endless debate areas for sure.

> If the functions in question were super complex, what you said would make total sense. For smaller get/post type functions, having niche interfaces still feels like an overkill.

I don't disagree, but the principle is very different from its enforcement.  There's no clear way to draw the line between "simple enough" and "complex enough" functions.  That just leads to fuzzy, haphazard implementation.  What _is_ enforceable is to adhere to one side or the other: always declare interfaces that match the needs of the function, or never do it.

Personally, I go with the former, because I feel strongly that it leads to better code at large scale: clearer cognitive boundaries, fewer arbitrary code injections, less hacks, better ease of testing.  Big interfaces, weak abstractions vs small interfaces, strong abstractions, and all that jazz.

Reply:
✍️ Thanks @ryanfkeepers